kotlin-native/NEW_MM.md

```diff
@@ -122,65 +116,52 @@ kotlin.targets.withType(KotlinNativeTarget::class.java) {
+    * Make sure that `execute` and `executeAfter` are only called on `Worker`s that were `Worker.start`ed or if the receiving `Worker` manually processes events with `Worker.processQueue`.
+
+We measured performance regressions with a slowdown up to a factor of 5. If you observe anything more significant, report to [this performance meta issue](https://youtrack.jetbrains.com/issue/KT-48526).
 
 ## Known bugs
 
-* Compiler caches are not supported, so compilation of debug binaries will be slower.
-* Freezing machinery is not thread-safe: if an object is being frozen on one thread, and its subgraph is being modified
-  on another, by the end the object will be frozen, but some subgraph of it might be not.
+* Compiler caches are not supported, so the compilation of debug binaries will be slower.
+* Freezing machinery is not thread-safe: if an object is being frozen on one thread, and its subgraph is being modified on another, by the end, the object will be frozen, but some subgraph of it might be not.
 * Documentation is not updated to reflect changes for the new MM.
-* There's no handling of application state on iOS: if application goes into the background, the collector will not be
-  throttled down; on the other hand the collection is not forced upon going into the background, which leaves
-  the application with a larger memory footprint than necessary, making it a more likely target to be terminated by the OS.
-* WASM (or indeed any target that does not have pthreads) is not supported with the new MM.
+* There's no application state handling on iOS: the collector will not be throttled down if the application goes into the background.
+  However, the collection is not forced upon going into the background, which leaves the application with a larger memory footprint than necessary, making it a more likely target to be terminated by the OS.
+* WASM (or any target that doesn't have threads) is not supported with the new MM.
+
 
 ## Workarounds
 
 ### Unexpected object freezing
-Some libraries might not be ready for the new memory model and freeze-transparency of `kotlinx.coroutines`, so unexpected `InvalidMutabilityException` 
-or `FreezingException` might appear.
-To workaround such cases, we added a `freezing` binary option that disables freezing fully (`disabled`) or 
-partially (`explicitOnly`).
-The former disables freezing mechanism at runtime (thus, making it no-op) while the latter disables automatic freezing of 
-[`@SharedImmutable`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.native.concurrent/-shared-immutable/) globals, 
-but keeps direct calls to [`freeze`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.native.concurrent/freeze.html) fully functional. 
+Some libraries might not be ready for the new memory model and freeze-transparency of `kotlinx.coroutines`, so unexpected `InvalidMutabilityException` or `FreezingException` might appear.
```

Comment:
_minor_: Sometimes we abbreviate the memory model to MM, sometimes we don't.

Reply:
Left  full 'memory model' only at the beginning